src/Adapter/File/ByteConvertor.php

```diff
@@ -0,0 +1,47 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\File;
+
+class ByteConvertor
+{
+    /**
+     * @var int
+     */
+    private $value;
+
+    public function __construct(int $value)
+    {
+        $this->value = $value;
+    }
+
+    public function fromMegaByteToByte(): int
+    {
+        return $this->value * 1048576;
+    }
```

Comment:
I'm not really sure about this class responsibility, I do understand its purpose but the API doesn't seem clear enough. We don't specify the unit of the input value but the only available method is `fromMegaByteToByte` so it implicitly means that it's supposed to be an input in MegaBytes which doesn't indicate the generic `$value` parameter

Also in that spirit shouldn't the class be named `MegaByteConvertor` then? But it wouldn't be generic enough I think.

Reply:
So an alternative I thought of was to use a private `__construct` with static build methods (naming is improvable):
- `buildWithMegaBytes`
- `buildWithKiloBytes`
- ...

And have some alternative public methods:
- `toBytes`
- `toKiloBytes`
- ...

Then it would fit its purpose of converting with more possibilities, what do you think?

Reply:
Oh and since it's a very simple class with no dependency I think it could be part of the Core namespace directly. In `PrestaShop\PrestaShop\Core\File\Converter` or in `PrestaShop\PrestaShop\Core\File` namespace probably

Reply:
Although the issue with this kind of implementation is that the class becomes stateful, it can only convert one value and you need to instantiate a new object for each value that needs to be converted.

I'm not sure it's the best approach for a tool class meant to convert data, sorry I'm commenting as I'm thinking about this 😅.

Reply:
So in the end I think of two possibilities:
- make this class a simple container of static methods like `fromMegaByteToByte(int $megaBytes): int` `fromByteToKiloByte(int $megaBytes): int` it makes the contract clear but it can add quite a lot of methods
- if we want to keep the idea of a stateful class maybe we should focus on the input/output units so a construct with `__construct(string $inputUnit, string $outputUnit)` with a few const, and a single `convert(int $value)` method

Reply:
I'm not sure to understand what you mean here A class from the Core namespace can be used in adapters handlers, pretty much from anywhere there's not reason to limit the Core classes into the Core namespace
Or maybe I didn't understand correctly what you mean here.

But if there's already `FileSizeConverter` existing why not use it or improve it if you need additional features? (currently it is based on string but you need some actual value conversion that could be added in the same class I think)